src/Resolvers/DomainTenantResolver.php

```diff
@@ -28,18 +28,34 @@ class DomainTenantResolver extends Contracts\CachedTenantResolver
 
     public function resolveWithoutCache(...$args): Tenant
     {
-        /** @var Domain $domain */
-        $domain = config('tenancy.domain_model')::where('domain', $args[0])->first();
+        $domain = $args[0];
 
-        if ($domain) {
-            static::$currentDomain = $domain;
+        /** @var Tenant|null $tenant */
+        $tenant = config('tenancy.tenant_model')::query()
+            ->whereHas('domains', function ($query) use ($domain) {
+                $query->select(['tenant_id', 'domain'])->where('domain', $domain);
```

Comment:
What's the select for?

Reply:
It's my habit to select only used columns.

To explain, why **you can** use `$tenant->domains->first()`:
The queries executed on line 34 looks like this:
```sql
select * from `tenants` where exists (select `tenant_id`, `domain` from `domains` where `tenants`.`id` = `domains`.`tenant_id` and `domain` = DOMAIN) limit 1
select * from `domains` where `domains`.`tenant_id` in (TENANT_ID) and `domain` = DOMAIN
```
`$tenant` **always** will have **only one** related domain (as we can see in the raw sql query), and it always will be the current domain (because we are finding tenants id by domain, then eager loading the current domain). I also checked that manually, by creating multiple tenants with multiple domains, and with cache enabled - it always returns correct domain model **without calling database**.

Reply:
Can I resolve this conversation or you don't like this approach? @stancl